src/main/kotlin/bd/clinic/modules/order/OrderService.kt

```diff
@@ -30,11 +41,39 @@ class OrderService(
     fun addPatientOrder(order: OrderDTO): OrderDTO {
         checkIfOrderNumberIsUnique(order.orderNumber)
         val patient = patientService.findPatientOrThrow(order.patientId)
+        sendPostRequestToAllLaboratories(order)
+
         return OrderDTO(orderRepository.save(order.toOrderEntity(patient)))
     }
 
+    private fun sendPostRequestToAllLaboratories(orderDTO: OrderDTO) {
+        LabServiceClient.infrastructurePortMap.forEach { (laboratoryId, port) ->
+            val infrastructureOrder = orderDTO.copy(
+                    examinations = orderDTO.examinations.filter { it.laboratoryId == laboratoryId })
+            if (infrastructureOrder.examinations.isNotEmpty())
+                labServiceClient.sendRequest(infrastructureOrder, port)
+        }
+    }
+
+    private fun sendGetRequestToAllLaboratories(orderDTO: OrderDTO): OrderDTO {
+        val examinationResultList = mutableListOf<ExaminationDTO>()
+        val laboratoriesList = orderDTO.examinations.map { it.laboratoryId }.distinct()
+
+        laboratoriesList.forEach {
+            labServiceClient.sendRequest(orderDTO.orderNumber, LabServiceClient.infrastructurePortMap.getValue(it))
+                    ?.examinations?.let { examResult -> examinationResultList.addAll(examResult) }
+        }
+
+        return orderDTO.copy(examinations = examinationResultList)
+    }
+
     private fun checkIfOrderNumberIsUnique(orderNumber: String) {
         if (orderRepository.existsByOrderNumber(orderNumber))
             throw OrderAlreadyExistsException(orderNumber)
     }
+
+    private fun checkIfOrderContainsAllExaminations(returnOrder: OrderDTO): Boolean {
+        return returnOrder.examinations.filter { it.isDone }.count() == returnOrder.examinations.size
+    }
```

Comment:
Zrobiłbym tak jak `checkIfOrderNumberIsUnique`, dzieki temu return w `getOrderByNumber` jest na samym końcu funkcji

Reply:
a to nic nie zwraca i tylko sprawdza, jesli wywali

Reply:
to zwraca prawdę, gdy zawiera wszystkie badania

Reply:
mowie jakby bylo jak juz zmienisz..

Reply:
ale ewentualnie mogę zmienić

Reply:
to zmien